### PR TITLE
fix: Build context backward compatability

### PR DESCRIPTION
**Pull request by gireesh-devtron**
# Description
Added backward compatibility for the build context.
Fixes https://github.com/devtron-labs/devtron/issues/3398

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ciConfig/CIDockerFileConfig.tsx [hunk 1]
```diff
@@ -206,11 +224,38 @@ export default function CIDockerFileConfig({
             </div>
         )
     }
+    const useRootBuildContextFlagFormState = currentCIBuildConfig?.useRootBuildContext
+    const [useRootBuildContextFlag,setUseRootBuildContextFlag] = useState<boolean>(useRootBuildContextFlagFormState)
+
+    const handleBuildContextCheckoutPathChange = (checkoutPath) => {
+        const val = checkoutPath.value
+        let flag = false
+        if(val === './'){
```
Comment (shivani170) [inline src/components/ciConfig/CIDockerFileConfig.tsx:233]: here as well

**Review by shivani170 — APPROVED** [review]

File: src/components/ciConfig/CIConfig.utils.ts [hunk 2]
```diff
@@ -173,6 +174,13 @@ export const getCIConfigFormState = (
                     : ciConfig?.ciBuildConfig?.dockerBuildConfig &&
                     ciConfig.ciBuildConfig.dockerBuildConfig?.buildContext),
             error: '', 
+        },
+        useRootBuildContext : {
+            value : (selectedCIPipeline?.isDockerConfigOverridden
+                ? selectedCIPipeline.dockerConfigOverride?.ciBuildConfig?.useRootBuildContext
+                : ciConfig?.ciBuildConfig &&
+                ciConfig.ciBuildConfig.useRootBuildContext),
```
Comment (sohel-rp) [inline src/components/ciConfig/CIConfig.utils.ts:182]: Can we change it to `ciConfig?.ciBuildConfig?.useRootBuildContext`

File: src/components/ciConfig/CIConfig.utils.ts [hunk 3]
```diff
@@ -436,6 +448,9 @@ export const getCIConfigDiffValues = (
     let globalBuildContextGitMaterialItem,currentBuildContextGitMaterialItem
     let globalBuildContext = globalCIConfig.ciBuildConfig?.dockerBuildConfig?.buildContext,
         currentBuildContext = ciConfigOverride?.ciBuildConfig?.dockerBuildConfig?.buildContext
+    let globalUseRootBuildContext = globalCIConfig.ciBuildConfig ? globalCIConfig.ciBuildConfig.useRootBuildContext : true
+    let currentUseRootBuildContext = ciConfigOverride?.ciBuildConfig ? ciConfigOverride?.ciBuildConfig.useRootBuildContext : true
```
Comment (sohel-rp) [inline src/components/ciConfig/CIConfig.utils.ts:452]: Is default as `true` expected?

Comment (gireesh-devtron) [reply src/components/ciConfig/CIConfig.utils.ts:452]: yes

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**